Monika After Story/game/script-story-events.rpy

```diff
@@ -1715,6 +1695,64 @@ label mas_notification_windowreact:
             m 5ttu "...right?"
     return
 
+init 5 python:
+    addEvent(
+        Event(
+            persistent.event_database,
+            eventlabel="mas_gift_giving_instructs",
+            conditional=(
+                "get_level() >= 15 "
+                "or mas_isSpecialDay() "
+                "and not persistent._mas_filereacts_historic"
+            ),
+            action=EV_ACT_QUEUE,
+            random=not persistent._mas_filereacts_historic
```

Comment:
Probably don't need this set as random. The queue action is sufficient and with this random condition, could technically get it very early on, regardless of special day or not.